radixdlt/babylon-wallet-android
# [ABW-3706] Destination account deposit rules

## Description
- load account deposit resource rules for non owned accounts on every asset addition.
- display error when deposit can't be made to an account due to it's deposit settings
- disable continue button if there are deposit errors
- if we can't load deposit rules (for example there is no network), condition based on the rules should be true - we allow continue button


## How to test

1. Set deny (all or for specific assets) deposit rule for account in Profile 1
2. Use Profile 2 and try to transfer any of assets from (1) to account in Profile 1, verify deposit warning is shown.
3. Verify that trying to build such transfer while having no network - Continue button is enabled after providing correct asset quantities.

## PR submission checklist
- [ ] I have tested that account deposit error is shown for resrouces
- [ ] I have tested that when failing to load deposit rules, user can proceed with the transfer
